<launch>

    <!-- 将 Urdf 文件的内容加载到参数服务器 -->
    
    <!-- 启动 gazebo -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch"/>   
 
 
    <!-- 在 gazebo 中显示机器人模型 -->
    <!-- 与rviz不同（4）：launch-->
    <node name="detector" pkg="rrt_exploration" type="frontier_opencv_detector.py" output="screen"/>
    <group ns="/robot_1">

    <include file="$(find big_map)/launch/includes/robot.launch.xml">
		<arg name="robot_name" value="robot_1"/>
		<arg name="init_pose" value="-x 0.0 -y 0.0 -z 0.0"/>
        <arg name="color" value="red"/>
	</include>

    <node name="dynamic" pkg="big_map" type="dynamic.py" output="screen">
	<param name="robot_name" value="robot_1"/>
    </node>

    <node name="map_odom" pkg="big_map" type="map_odom.py" output="screen" >
    <param name="frame_id" value="robot_1" />
    <param name="child_frame_id" value="robot_1" />
    </node>

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" >
    <param name="publish_frequency" type="double" value="30.0" />
    <param name="use_tf_static" type="bool" value="false" />
    </node>
    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" /> 

    <include file="$(find big_map)/launch/includes/move_base.launch">
		<arg name="namespace" value="robot_1"/>
	</include> 
    
    <node name="canny" pkg="big_map" type="frontier_opencv_detector.py" output="screen" >
    <param name="map_topic" value="map"/>
    <param name="color_r" value="255"/>
    <param name="color_g" value="0"/>
    <param name="color_b" value="0"/>
    </node>
    </group>

    <group ns="/robot_2">
   
     <include file="$(find big_map)/launch/includes/robot.launch.xml">
		<arg name="robot_name" value="robot_2"/>
		<arg name="init_pose" value="-x 2.0 -y 2.0 -z 0.0"/>
        <arg name="color" value="green"/>
	</include>
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" >
    <param name="publish_frequency" type="double" value="30.0" />
    <param name="use_tf_static" type="bool" value="false" />
    </node>
    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" /> 

    <node name="dynamic" pkg="big_map" type="dynamic.py" output="screen">
	<param name="robot_name" value="robot_2"/>
    </node>
    
    <node name="map_odom" pkg="big_map" type="map_odom.py" output="screen" >
    <param name="frame_id" value="robot_2" />
    <param name="child_frame_id" value="robot_2" />
    </node>
    <node name="canny" pkg="big_map" type="frontier_opencv_detector.py" output="screen" >
    <param name="map_topic" value="map"/>
    <param name="color_r" value="0"/>
    <param name="color_g" value="255"/>
    <param name="color_b" value="0"/>
    </node>
        <include file="$(find big_map)/launch/includes/move_base.launch">
		<arg name="namespace" value="robot_2"/>
	</include> 
    
    
    </group>

    <group ns="/robot_3">

     <include file="$(find big_map)/launch/includes/robot.launch.xml">
		<arg name="robot_name" value="robot_3"/>
		<arg name="init_pose" value="-x -2.0 -y -2.0 -z 0.0"/>
        <arg name="color" value="blue"/>
	</include>
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" >
    <param name="publish_frequency" type="double" value="30.0" />
    <param name="use_tf_static" type="bool" value="true" />
    </node>
    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" /> 

    <node name="dynamic" pkg="big_map" type="dynamic.py" output="screen">
	<param name="robot_name" value="robot_3"/>
    </node>
    
    <node name="map_odom" pkg="big_map" type="map_odom.py" output="screen" >
    <param name="frame_id" value="robot_3" />
    <param name="child_frame_id" value="robot_3" />
    </node>
    <node name="canny" pkg="big_map" type="frontier_opencv_detector.py" output="screen" >
    <param name="map_topic" value="map"/>
    <param name="color_r" value="0"/>
    <param name="color_g" value="0"/>
    <param name="color_b" value="255"/>
    </node>
        <include file="$(find big_map)/launch/includes/move_base.launch">
		<arg name="namespace" value="robot_3"/>
	</include> 
    
    
    </group>
    
  
    
    
    <include file="$(find big_map)/launch/includes/initposes.launch"/>
    <include file="$(find big_map)/launch/includes/map_merge.launch"/>    

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find big_map)/rviz/three.rviz" >
    <remap from="move_base_simple/goal" to="robot_1/move_base_simple/goal"/>
    </node>


</launch>